<?xml version="1.0"?>
 
 <launch>  
      
   <!-- Laser to Pointcloud -->
  <node pkg="vigir_laserscan_to_pointcloud" type="laserscan_to_pointcloud_node" name="scan_to_clouds_pub_node" output="screen" ns="lidar">
    <remap from="scan" to="/most_intense"/>
    <remap from="scan_cloud" to="/lidar/scan_cloud"/>
    <param name="use_high_fidelity_projection" value="true"/>
    <param name="target_frame" value="spin_lidar_lidar_mount_link"/>
  </node>
  
  <!-- Laser Frame -->
  <!--  <node pkg="tf" type="static_transform_publisher" name="$(arg sensor_identifier)_laser_frame_pub"          args="0 0 0 0 -1.57 0 $(arg  sensor_identifier)_spinning_lidar_spin_link $(arg sensor_identifier)_laser_frame 100" />-->
  
  <!-- Actuator Frame -->
  <node pkg="tf" type="static_transform_publisher" name="lidar_actuator_frame_pub" args="-0.3 0.1 0.58 0 -1.57 0 base_link lidar_actuator_frame 100" />
  
  <!-- Aggregator -->
   <node pkg="lidar_calibration" type="cloud_aggregator_node" name="cloud_aggregator_node" output="screen" ns="lidar">
    <remap from="cloud" to="scan_cloud"/>
    <param name="target_frame" value="lidar_actuator_frame"/>
    <param name="rotations" value="1" />
  </node> 
  
  <!-- Calibration -->
  <node pkg="lidar_calibration" type="lidar_calibration_node" name="lidar_calibration_node" output="screen" ns="lidar_calibration" args="--m">
    <param name="actuator_frame" value="lidar_actuator_frame"/>
    <param name="rotation_offset_pitch" value="0.0"/>
    
    <param name="detect_ground_plane" value="false"/>
    <param name="detect_ceiling" value="false"/>
    <param name="ground_frame" value="base_link"/>
    
    <param name="o_spin_frame" value="spin_lidar_lidar_mount_link"/>
    <param name="o_laser_frame" value="spin_lidar_link_hokuyo_laser_frame"/>
    <param name="tf_wait_duration" value="10"/>
     
    <param name="save_calibration" value="true"/>
    <param name="save_path" value="$(find lidar_calibration)/calibrations/hector_tracker_lidar_calibration.urdf.xacro"/>
    
    <remap from="request_scans" to="/lidar/request_scans" />
    <remap from="reset_clouds" to="/lidar/reset_clouds" />
  </node>
</launch>
